packages/cdktf/lib/terraform-functions.ts

```diff
@@ -0,0 +1,1250 @@
+    return asBoolean(terraformFunction("anytrue", [listOf(anyValue)])(value));
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/chunklist.html chunklist} splits a single list into fixed-size chunks, returning a list of lists.
+   * @param {Array} value
+   * @param {number} chunkSize
+   */
+  public static chunklist(value: any[], chunkSize: number | IResolvable) {
+    return asList(
+      terraformFunction("chunklist", [listOf(anyValue), numericValue])(
+        value,
+        chunkSize
+      )
+    );
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/coalesce.html coalesce} takes any number of arguments and returns the first one that isn't null or an empty string.
+   * @param {Array} value - Arguments are passed in an array
+   */
+  public static coalesce(value: any[]) {
+    return terraformFunction("coalesce", listOf(anyValue))(...value);
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/coalescelist.html coalescelist} takes any number of list arguments and returns the first one that isn't empty.
+   * @param Array} value - Arguments are passed in an array
+   */
+  public static coalescelist(value: any[][]) {
+    return asList(
+      terraformFunction("coalescelist", listOf(anyValue))(...value)
+    );
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/compact.html compact} takes a list of strings and returns a new list with any empty string elements removed.
+   * @param {Array} value
+   */
+  public static compact(value: string[]) {
+    return asList(terraformFunction("compact", [listOf(anyValue)])(value));
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/concat.html concat} takes two or more lists and combines them into a single list.
+   * @param {Array} value
+   */
+  public static concat(value: any[][]) {
+    return asList(terraformFunction("concat", listOf(anyValue))(...value));
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/contains.html contains} determines whether a given list or set contains a given single value as one of its elements.
+   * @param {Array} list
+   * @param {any} value
+   */
+  public static contains(list: any[], value: any) {
+    return asBoolean(
+      terraformFunction("contains", [listOf(anyValue), anyValue])(list, value)
+    );
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/distinct.html distinct} takes a list and returns a new list with any duplicate elements removed.
+   * @param {Array} list
+   */
+  public static distinct(list: any[]) {
+    return asList(terraformFunction("distinct", [listOf(anyValue)])(list));
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/element.html element} retrieves a single element from a list.
+   * @param {Array} list
+   * @param {number} index
+   */
+  public static element(
+    list: any[] | string | IResolvable,
```

Comment:
Does this work for strings?

I noticed that other collection functions don't take `IResolvable`; shouldn't any array parameter support using `IResolvable`?